### ISSUE TITLE
Collectives - Update verbatim examples to example environment

### ISSUE THREAD

**Issue by wesbland** [mpi-forum/mpi-issues#626]
# Problem

The grey box formatting around examples (added in MPI-4.0) raises that many examples use verbatim instead of the example environment, which means the example is not formatted consistently.  This seems like something for the MPI-4.1 cleanup efforts.

# Proposal

Review example listings without a grey box and update to use correct `\begin{example}...\end{example}` formatting.

# Changes to the Text

Update LaTeX throughout the document, e.g., Section 7.6.3 Example 1: "Three-Group Pipeline".

# Impact on Implementations

None.

# Impact on Users

None.

**Comment by schulzm** [comment]
This seems to be done for the whole chapter already - however, in many cases, examples get interrupted by floating figures, which looks odd - I wonder if there is a way to prevent this

**Comment by wgropp** [comment]
It is hard to automatically ensure figure placement doesn't interrupt flowed text, which is how the examples are presented.  However, I can work to move the figures around if there are particularly bad examples.

**Comment by schulzm** [comment]
Yes, that is the usual problem - let's put that on the list to check for in the next RC reading. As far as I am concerned, the ticket could be closed, but @tonyskjellum should probably agree to this first.

**Comment by tonyskjellum** [comment]
let me check... I will look at ticket today.

**Comment by wesbland** [comment]
This wasn't read at the December 2022 meeting. The last opportunity for MPI 4.1 is to have it ready at the March 2022 meeting (and it needs to "pass" the reading).

@tonyskjellum Are you (or is someone else) planning to push this forward or should we move it out of the plan for MPI 4.1?

**Comment by bkmgit** [comment]
This seems to be done.

**Comment by wesbland** [comment]
Agree with @bkmgit
